### PR TITLE
Move testing documentation out of root README into docs folder

### PR DESCRIPTION
**Pull request by aaron-collier**
The first of a few documentation PRs related to #265 and #288

- [x] Moves the existing testing content from README to `docs/testing.md`
- [x] Adds framework introduction to `docs/testing.md`
- [x] Adds package testing instructions to `docs/testing.md`
- [x] Adds link to `docs/testing.md` to `docs/README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,17 +104,8 @@ $ curl -F "upload=@myfile.pdf;type=application/pdf" http://localhost:8080/v1/fil
...
-$ go test -v ./... -short
-```
-
-The integration test depends on the taco binary and [Localstack](docs/localstack.md) running.  Once these conditions are met you can run the integration tests using:
-
-```shell
-$ go test test/integration_test.go
-```
 
+Testing and test driven development is closely tied to the entire technology stack. Review the [Testing Guidelines & Framework](./docs/testing.md) documentation.
```
Comment (jcoyne) [inline README.md]: I'm not sure that "closely tied to the entire technology stack" adds any help here. Can we remove?

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Testing Guidelines & Framework
+
+## Testing Framework
+
+[Go](https://golang.org) has extensive support for automated [testing](https://golang.org/pkg/testing/) built into the language.
+
+
+Additionally, we utilize the [Gofight](https://github.com/appleboy/gofight) framework to test API handler mock responses.
```
Comment (jcoyne) [inline docs/testing.md]: Do you want to mention `gopkg.in/h2non/baloo.v3` for end-to-end tests?

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```shell
+$ go test -v ./... -short
+```
+
+## Go Package Tests
+The package tests are dependent on [Localstack](docs/localstack.md) running as they are testing individual connectivity to
+external services.
+
+```shell
+AWS_REGION=localstack go test -v ./[PACKAGE NAME]/
```
Comment (jcoyne) [inline docs/testing.md]: I believe the only package that fits this criteria right now is `db`

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+```
+&awserr.requestError{awsError:(*awserr.baseError)(0xc420060a00), statusCode:400, requestID:"8e442552-4370-46d1-8226-73a0db496dcd"}
+```
+
+With a corrisponding
+```
+WARNING:localstack.services.dynamodb.dynamodb_listener: Unknown table: resources not found in {}
+```
+
+Is an indication that the particular [Localstack](docs/localstack.md) resource is missing. Follow the instructions in the document
```
Comment (jcoyne) [inline docs/testing.md]: replace "resource" with "dynamodb table"

File: docs/testing.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+## Troubleshooting Common Errors
+
+### Missing Region
+
+```
+&awserr.baseError{code:"MissingRegion", message:"could not find region configuration", errs:[]error(nil)}
+```
+
+This error indicates that the command line aws region key (`AWS_REGION`) are missing.
```
Comment (jcoyne) [inline docs/testing.md]: replace "command line" with "environment variable"

File: docs/testing.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+$ go test -v ./... -short
+```
+
+## Go Package Tests
+
+_Note: The db package test is dependent on [Localstack](docs/localstack.md) running as they are testing individual connectivity to
+external services._
+
+```shell
+AWS_REGION=localstack go test -v ./[PACKAGE NAME]/
```
Comment (eefahy) [inline docs/testing.md]: This makes sense, however I'm a little confused as to how to know programmatically what packages I need to run. I suppose one option is to just change our testing pattern for every package that we add but that feels a little fragile.

Comment (jcoyne) [reply docs/testing.md]: How about:
```
go list -f '{{if len .TestGoFiles}}"go test {{.ImportPath}}"{{end}}' ./...|xargs -L 1 sh -c
```

Comment (eefahy) [reply docs/testing.md]: Is that really want we want? I have more questions about why something like package tests (unit tests) need an external service running? Can we mock ext services for those?

**Pull request merged**